### PR TITLE
Some CRT implementations

### PR DESCRIPTION
**Pull request by encounter**
Implements `_initterm`, `_initterm_e`, `malloc` and `free`.

Updates `Machine::call_x86` to return EAX.

### CODE REVIEW

File: win32/src/winapi/ucrtbase.rs [hunk 1]
```diff
@@ -6,12 +6,37 @@ use crate::Machine;
...
+    if (end - start) % 4 != 0 {
+        panic!("unaligned _initterm_e");
+    }
+    let slice = machine.mem().sub(start, end - start).as_slice_todo();
+    let slice =
+        unsafe { std::slice::from_raw_parts(slice.as_ptr() as *const u32, slice.len() / 4) };
+    for &addr in slice {
+        if addr != 0 {
+            let err = machine.call_x86(addr, vec![]).await;
+            if err != 0 {
```
Comment (evmar) [inline win32/src/winapi/ucrtbase.rs:35]: Is it intentional this doesn't match initterm?

I looked at the win2k msvcrt.dll I had but it only has initterm, no _e variant, and its behavior matches the function you wrote.

File: x86/src/x86.rs [hunk 2]
```diff
@@ -143,7 +143,7 @@ impl std::future::Future for X86Future {
         let cpu = self.cpu;
         let cpu = unsafe { &mut *cpu };
         if cpu.regs.get32(Register::ESP) == self.esp {
-            std::task::Poll::Ready(())
+            std::task::Poll::Ready(cpu.regs.get32(Register::EAX))
```
Comment (evmar) [inline x86/src/x86.rs:146]: Wow so easy!

File: win32/src/winapi/ucrtbase.rs (hunk 1 above)
Comment (encounter) [reply win32/src/winapi/ucrtbase.rs:35]: Yes, `_initterm` calls `PVFV` (`void (*)(void)`) functions, whereas `_initterm_e` calls `PIFV` (`int (*)(void)`) functions, and the latter short circuits on a non-zero return value.

Comment (encounter) [reply win32/src/winapi/ucrtbase.rs:35]: (It's totally possible this behavior has changed over time though)

**Pull request merged**